Makefile

```diff
@@ -443,16 +516,149 @@ doc_diagram_sources+=$(addprefix src/lib/transition_frontier/res/,*.dot *.tex *.
+		./scripts/export-git-env-vars.sh
+
+########################################
+# Debian packages
+
+# Helper function for building Debian packages
+define build_debian_package
+	BUILD_DIR="${PWD}/_build" \
+	DUNE_PROFILE=$(DUNE_PROFILE) \
+	MINA_DEB_CODENAME="$(CODENAME)" \
+	./scripts/debian/build.sh $(1)
+endef
+
+
+.PHONY: debian-build-logproc
+debian-build-logproc: ## Build the Debian logproc package
+	$(call build_debian_package,logproc)
+
+.PHONY: debian-build-daemon
+debian-build-daemon: ## Build the Debian daemon package
+	$(call build_debian_package,daemon)
+	
+.PHONY: debian-build-daemon-devnet
+debian-build-daemon-devnet: ## Build the Debian daemon package for devnet
+	$(call build_debian_package,daemon_devnet)
+
+.PHONY: debian-build-daemon-mainnet
+debian-build-daemon-mainnet: ## Build the Debian daemon package for mainnet
+	$(call build_debian_package,daemon_mainnet)
+
+.PHONY: debian-build-archive
+debian-build-archive: ## Build the Debian archive package
+	$(call build_debian_package,archive_berkeley)
+		
+.PHONY: debian-build-archive-devnet
+debian-build-archive-devnet: ## Build the Debian archive package for devnet
+	$(call build_debian_package,archive_devnet)
+
+.PHONY: debian-build-archive-mainnet
+debian-build-archive-mainnet: ## Build the Debian archive package for mainnet
+	$(call build_debian_package,archive_mainnet)
+
+.PHONY: debian-build-rosetta
+debian-build-rosetta: ## Build the Debian Rosetta package
+	$(call build_debian_package,rosetta_berkeley)
+
+.PHONY: debian-build-rosetta-devnet
+debian-build-rosetta-devnet: ## Build the Debian Rosetta package for devnet
+	$(call build_debian_package,rosetta_devnet)
+
+.PHONY: debian-build-rosetta-mainnet
+debian-build-rosetta-mainnet: ## Build the Debian Rosetta package for mainnet
+	$(call build_debian_package,rosetta_mainnet)
+
 ########################################
 # Docker images
 
+.PHONY: start-local-debian-repo
+start-local-debian-repo: ## Start a local Debian repository
+	./scripts/debian/aptly.sh stop || true
+
+	./scripts/debian/aptly.sh start \
+		--codename $(CODENAME) \
+		--debians _build \
+		--component unstable \
+		--clean \
+		--background \
+		--wait
+	
+
 .PHONY: docker-build-toolchain
 docker-build-toolchain: ## Build the toolchain to be used in CI
-	./scripts/docker/build.sh \
+	BUILD_DIR=./_build \
+		./scripts/docker/build.sh \
 		--deb-codename $(CODENAME) \
 		--service mina-toolchain \
 		--version mina-toolchain-$(CODENAME)-$(GITHASH)
 
+# General function for building Docker images
+define build_docker_image
+	BUILD_DIR=./_build \
+	MINA_DEB_CODENAME=$(CODENAME) \
+	KEEP_MY_TAGS_INTACT=true \
+	. ./scripts/export-git-env-vars.sh \
+	&& ./scripts/docker/build.sh \
+		--deb-codename $(CODENAME) \
+		--service $(1) \
+		--version "$$MINA_DEB_VERSION" \
+		--branch "$$GITBRANCH" \
+		--network $(2) \
+		--no-cache
+
+	./scripts/debian/aptly.sh stop
+endef
+
+.PHONY: docker-build-daemon
+docker-build-daemon: SHELL := /bin/bash
```

Comment:
Why do you need this extra variable?

Reply:
I learn that by default makefile is using sh. I didn't want to change it globally, just for scripts that are using bash scripts